share/functions/__fish_clang_complete.fish

```diff
@@ -0,0 +1,14 @@
+# This function is compatible with clang, clang++, and variations thereof, and is shared
+# by clang.fish and clang++.fish.
+# We dynamically query the head at `(commandline -o)[1]` to get the correct completions.
+function __fish_clang_complete
+	# If the result is for a value, clang only prints the value, so
+	# completions for `-std=` print `c++11` and not `-std=c++11` like we need. See #4174.
+	set -l prefix ""(string replace -fr -- '^(.*=)[^=]*' '$1' (commandline -ct))
```

Comment:
> This `""` is entirely useless. It does literally nothing. What do you expect it to do?

I was having a hell of a hard time getting a valid value here and thought perhaps it was the whole "null variables expand to no argument rather than empty argument unless quoted", so at some point I added the quotes and left them in thereafter.